From 4c0ee3677998b131954a6e67b3faf614c38d3670 Mon Sep 17 00:00:00 2001
From: Simon Denier <simon@sogilis.com>
Date: Thu, 19 Jul 2012 11:41:16 +0200
Subject: [PATCH 3/3] Implement missing integer comparison operations

---
 lua/forth |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/lua/forth b/lua/forth
index 95d0628..f84c156 100755
--- a/lua/forth
+++ b/lua/forth
@@ -39,6 +39,26 @@ symbol_table['.'] = function(...)
   io.write(tostring(pop()), " ")
 end -- pop
 
+symbol_table['<='] = function(...)
+  local a, b = pop(2)
+  push(a <= b)
+end -- less/equal
+
+symbol_table['<'] = function(...)
+  local a, b = pop(2)
+  push(a < b)
+end -- less
+
+symbol_table['>='] = function(...)
+  local a, b = pop(2)
+  push(a >= b)
+end -- greater/equal
+
+symbol_table['>'] = function(...)
+  local a, b = pop(2)
+  push(a > b)
+end -- greater
+
 symbol_table.IF = function(original_dispatcher)
   if pop() then
     return make_dispatch_execute_until(original_dispatcher, "ELSE",
-- 
1.7.10.2 (Apple Git-33)

