From cc48df8ba5e034e57bc1424ebb976702d999a92d Mon Sep 17 00:00:00 2001
From: Simon Denier <simon@sogilis.com>
Date: Thu, 19 Jul 2012 11:41:16 +0200
Subject: [PATCH 3/3] Implement missing integer comparison operations

---
 forth | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/forth b/forth
index 28af5ce..b76fb2c 100755
--- a/forth
+++ b/forth
@@ -49,6 +49,26 @@ symbol_table['.'] = function(...)
   io.write(tostring(pop()), " ")
 end -- '.'
 
+symbol_table['<='] = function(...)
+  local b, a = pop(), pop()
+  push(a <= b)
+end -- less/equal
+
+symbol_table['<'] = function(...)
+  local b, a = pop(), pop()
+  push(a < b)
+end -- less
+
+symbol_table['>='] = function(...)
+  local b, a = pop(), pop()
+  push(a >= b)
+end -- greater/equal
+
+symbol_table['>'] = function(...)
+  local b, a = pop(), pop()
+  push(a > b)
+end -- greater
+
 symbol_table.IF = function(original_dispatcher)
   if pop() then
     return make_dispatch_execute_until(original_dispatcher, "ELSE",
-- 
1.8.1.3

